### PR TITLE
AWS reuse_cluster by test_id tag, and not via ip addresses

### PR DESCRIPTION
**Pull request by fruch**
addresses #836

### CODE REVIEW

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -253,8 +253,14 @@ def _create_mixed_instances(self, count, interfaces, ec2_user_data, dc_idx, tags
...
 
     def _get_instances(self, dc_idx):
+
+        test_id = self.params.get('test_id', default=None)
+        if not test_id:
+            raise ValueError("test_id should be configured for using reuse_cluster")
+
         ec2 = ec2_client.EC2Client(region_name=self.region_names[dc_idx])
-        return [ec2.get_instance_by_private_ip(ip) for ip in self._node_private_ips]
+        instances = list_instances_aws(tags_dict={'TestId': test_id}, region_name=self.region_names[dc_idx])
```
Comment (bentsi) [inline sdcm/cluster_aws.py]: how do you handle different types of nodes (Loader, Monitor)?

Comment (fruch) [reply sdcm/cluster_aws.py]: a very good question...

Comment (fruch) [reply sdcm/cluster_aws.py]: @bentsi no I've added a NodeType and NodeIndex flags, and I'm sorting by the NodeIndex

**Comment by roydahan** [comment]
@fruch / @bentsi, did we add to documentation?
Worth even an email to QA explaining how to use.

**Comment by bentsi** [comment]
@fruch did you add this to doc? 
@roydahan we will explain on the QA meeting and send and email

**Comment by roydahan** [comment]
backported by me.
I still don't see a documentation on how to use it.

**Comment by fruch** [comment]
@roydahan seee #852 for the docs

**Pull request merged**